ledger/src/blockstore_db.rs

```diff
@@ -802,6 +802,12 @@ pub trait ColumnIndexDeprecation: Column {
     }
 }
 
+/// Helper trait to transition a column between bincode formats
+pub trait BincodeTypeTransition: Column + TypedColumn {
+    /// This should have a different serialized size than the TypedColumn::Type
+    type OldType: Serialize + DeserializeOwned + Into<Self::Type>;
+}
+
```

Comment:
This type trait and its associated code is pretty verbose and actually adding more complexity here.

Might be simple just to do something like below:
```diff
diff --git a/ledger/src/blockstore_db.rs b/ledger/src/blockstore_db.rs
index b65df82ee0..eebf163400 100644
--- a/ledger/src/blockstore_db.rs
+++ b/ledger/src/blockstore_db.rs
@@ -1586,7 +1586,7 @@ where
     }
 }
 
-impl<C> LedgerColumn<C>
+impl<C: 'static> LedgerColumn<C>
 where
     C: TypedColumn + ColumnName,
 {
@@ -1635,7 +1635,12 @@ where
             &self.read_perf_status,
         );
         if let Some(pinnable_slice) = self.backend.get_pinned_cf(self.handle(), key)? {
-            let value = deserialize(pinnable_slice.as_ref())?;
+            use std::any::TypeId;
+            let value = deserialize(pinnable_slice.as_ref()).or_else(|err| {
+                if TypeId::of::<C>() == TypeId::of::<columns::MerkleErasureMeta>() {
+                    // retry with the old erasure meta type.
+                    // ...
+                }
+                Err(err)
+            })?;
             result = Ok(Some(value))
         }
 
```

Reply:
I added it to provide an easy way to transition types of already existing columns in blockstore. Similar to ColumnIndexDeprecation.

All you have to do is implement the trait and TypedColumn. And after the old type is no longer in blockstore, you can just remove the trait impl and the associated migration test.

Figured its worth adding the complexity here to make future transitions frictionless.